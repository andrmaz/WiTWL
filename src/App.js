/* eslint-disable max-len */
/* eslint-disable quotes */

import React from 'react';
import { apiCall } from './api/apiCall';
import Search from './component/Search/Search';
import Current from './component/Current/Current';
import Forecast from './component/Forecast/Forecast';
import Location from './component/Location/Location';
import './App.scss';

class App extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            city: '',
            weather: {},
        };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    };

    handleChange(event){
        this.setState({ city: event.target.value });
    };

    async handleSubmit(event){
        event.preventDefault();
        const data = await apiCall(this.state.city);
        this.setState({ weather: data});
    };

    render() {
        const { weather } = this.state;
        return (
            <div className='container'>
                <main>
                    <Search
                        city={this.state.city}
                        handleChange={this.handleChange} 
                        handleSubmit={this.handleSubmit} 
                    />
                    <Location weather={weather} />
                    <Current weather={weather} />
                    <Forecast weather={weather} />
                </main> 
                <a href="https://icons8.com/icon/D2RKDghvz9nr/tempo-piovoso">Tempo piovoso icon by Icons8</a>
            </div>
        );
    };
};

export default App;
