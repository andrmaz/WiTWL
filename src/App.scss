$body-color: whitesmoke;

.container {
  background-color: $body-color;
  height: 100%;
  width: 100%;
  display: flex;

  main {
    height: 80%;
    width: 80%;
    display: grid;
    margin: auto;
    box-shadow: 3px 3px 5px 6px #ccc;
    border-radius: 5px;
    grid-template-columns: 45% 55%;
    grid-template-rows: 10% 50% 40%;
  }

  a {
    position: absolute;
    right: 5px;
    bottom: 5px;
  }
}