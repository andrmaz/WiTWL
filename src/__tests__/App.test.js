import React from 'react'
import { shallow } from 'enzyme'
import App from '../App'

let wrapper
let instance
let event
const mockWeather = jest.mock()

describe('App Component', () => {
    beforeEach(() => {
        wrapper = shallow(<App />)
        instance = wrapper.instance()
        expect.hasAssertions()
    })
    it('should render Component correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
    it('should be a stateful component', () => {
        expect(instance).toBeInstanceOf(App)
    })
    it('should render initial State correctly', () => {
        expect(wrapper.state().city).toEqual('')
        expect(wrapper.state().weather).toEqual({})
    })
    it('should set city correctly', () => {
        wrapper.setState({ city: 'New York' })
        expect(wrapper.state().city).toEqual('New York')
    })
    it('should set weather correctly', () => {
        wrapper.setState({ weather: mockWeather })
        expect(wrapper.state().weather).toEqual(mockWeather)
    })
    it('should handle changes correctly', () => {
        event = { target: { value: 'London'}}
        instance.handleChange(event)
        expect(wrapper.state().city).toEqual('London')
    })
    it('should handle submit correctly', async () => {
        event = { preventDefault: () => {} }
        const mockSubmit = jest.spyOn(event, 'preventDefault')
        instance.handleSubmit(event)
        expect(mockSubmit).toHaveBeenCalledTimes(1)
    })
})

