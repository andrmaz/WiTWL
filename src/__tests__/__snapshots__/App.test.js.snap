// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render Component correctly 1`] = `
<div
  className="container"
>
  <main>
    <Search
      city=""
      handleChange={[Function]}
      handleSubmit={[Function]}
    />
    <Location
      weather={Object {}}
    />
    <Current
      weather={Object {}}
    />
    <Forecast
      weather={Object {}}
    />
  </main>
  <a
    href="https://icons8.com/icon/D2RKDghvz9nr/tempo-piovoso"
  >
    Tempo piovoso icon by Icons8
  </a>
</div>
`;
